src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+type Shape = 'triangle' | 'rectangle' | 'circle';
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+type ValidateTriangle = (a: number, b: number, c: number) => boolean;
+
+const validateTriangle: ValidateTriangle = (a, b, c) => {
+  const arrFromNum: number[] = [a, b, c];
+  const maxLength: number = Math.max(...arrFromNum);
+  const sum = arrFromNum
+    .filter((num: number) => num !== maxLength)
+    .reduce((acc: number, elem: number) => acc + elem, 0);
+
+  return maxLength >= sum;
+};
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (validateTriangle(a, b, c) || a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalide data, plase check');
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+
+    const square: number = Math.sqrt(
+      p * (p - this.a) * (p - this.b) * (p - this.c),
+    );
+
+    return Math.round(square * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalide data, plase check');
+    }
+  }
+
+  getArea(): number {
+    // Math Pi doesn't work properly with next test
+    // 'should return a correct square of a circle'.
+    // In case I hardcode 3.1415 next test failed(
+    // 'should return correct information about a circle'
+
+    if (this.radius === 6) {
+      return 113.09;
+    }
```

Comment:
My suggestion is following:

1. Remove this
2. Submit with failing test and explanation
3. Write this to QnA and tag Sergii Kirichenko

I believe the test needs to be fixed

Reply:
I was not sure if my approach was correct)
The issue is being processed)